### ISSUE TITLE
Malformed json/yaml model file in discoverDomain due to with  newline handling 

### ISSUE THREAD

**Issue by rssh22** [oracle/weblogic-deploy-tooling#999]
As of #829 this should be solved. However, the problem happens again in release 1.9.18. 

@Bneve5 @rssh22 it appears that the core issue was the embedded newline characters in the string value.  I have created a PR that corrects the embedded newline handling that I believe will fix this issue.

_Originally posted by @robertpatrick in https://github.com/oracle/weblogic-deploy-tooling/issues/823#issuecomment-771307630_

**Comment by robertpatrick** [comment]
@rssh22 I am not seeing the PR.  Are you sure you created it on this repo?

**Comment by rssh22** [comment]
@robertpatrick I did not create the PR. I guess that you did because of your comment in https://github.com/oracle/weblogic-deploy-tooling/issues/823#issuecomment-771307630_

As you can see in the release notes for 1.9.9 this bug was solved there:
![image](https://user-images.githubusercontent.com/25611683/141154028-49bbc369-f67b-4aaa-bcb6-894fbe535da1.png)

**Comment by robertpatrick** [comment]
> @Bneve5 @rssh22 it appears that the core issue was the embedded newline characters in the string value. I have created a PR that corrects the embedded newline handling that I believe will fix this issue.

@rssh22 OK, your previous comment included an unquoted reference to a comment I made 9 months ago so I read it as your comment.

**Comment by rssh22** [comment]
Here is another example of malformed json generated by the tool:

```
            "siloe-web-jee-r01a-iq-jeec-jds-u0" : {
                "Target" : "wlc-pre-32-c1",
                "JdbcResource" : {
                    "JDBCConnectionPoolParams" : {
                        "MaxCapacity" : 15,
                        "WrapTypes" : "true",
                        "RemoveInfectedConnections" : "true",
                        "StatementCacheType" : "LRU",
                        "InactiveConnectionTimeoutSeconds" : 20,
                        "StatementTimeout" : 60,
                        "HighestNumWaiters" : 30,
                        "ShrinkFrequencySeconds" : 900,
                        "StatementCacheSize" : 10,
                        "TestTableName" : "SQL SELECT 1 FROM DUAL",
                        "ConnectionReserveTimeoutSeconds" : 10,
                        "CapacityIncrement" : 1,
                        "IgnoreInUseConnectionsEnabled" : "true",
                        "InitialCapacity" : 1,
                        "TestFrequencySeconds" : 120,
                        "LoginDelaySeconds" : 0,
                        "TestConnectionsOnReserve" : "true",
                        "SecondsToTrustAnIdlePoolConnection" : 10,
                        "InitSql" : "SQL BEGIN

EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_SORT = SPANISH_AI';

EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_COMP = LINGUISTIC';

EXECUTE IMMEDIATE 'ALTER SESSION SET \"_b_tree_bitmap_plans\" = FALSE';

EXECUTE IMMEDIATE 'ALTER SESSION SET \"star_transformation_enabled\" = FALSE';
END;",
                        "ConnectionCreationRetryFrequencySeconds" : 0
                    },

```

**Comment by robertpatrick** [comment]
@rssh22 This JSON output formatting is a different problem, not related to the YAML parser issues surrounding newlines.  I will ask someone to look into this...

**Comment by rssh22** [comment]
I though it was the same cause.  Thank you @robertpatrick.

**Comment by robertpatrick** [comment]
PR #1011 should fix the JSON output formatting issue, once merged and released.  Since most people are using YAML, this was an oversight on our part.  The JSON parser was already properly "de-escaping" the special characters but the code generating the JSON output was not properly escaping them.

**Comment by ddsharpe** [comment]
Merge #1101 partially fixes this issue.  Another change is pending.

**Comment by robertpatrick** [comment]
I believe that this issue was about the output formatting only so closing this issue as resolved by PR #1011. 

In WDT 2.0, we are replacing the YAML parser and output formatter with snakeyaml.  This will cause some minor changes in what the input YAML needs to look like as well as how the YAML output is formatted (hence the 2.0 version) but we hope it will provide a better all-around YAML experience for our users.
